ext_py/tests/test_fp_vector_slices.py

```
@@ -0,0 +1,293 @@
+    assert list(sm) == [2, 4]
+
+
+def test_stale_slice_handles_raise_python_exception_after_parent_shrink():
+    v = fp.FpVector.from_py(5, [1, 2, 3, 4])
+    s = v.const[1:4]
+    sm = v.mut[1:4]
+
+    v.set_scratch_vector_size(2)
+
+    with pytest.raises(IndexError):
+        s.prime
```

Ruff (0.15.20)

[warning] Found useless expression. Either assign it to a variable or remove it.

(B018)